src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -269,12 +288,25 @@ public int getAutoStopAfter() {
         }
 
         public void setAutoStopAfter(int autoStopAfter) {
-            if (autoStopAfter < 0)
-                throw new IllegalArgumentException("Timer for automatically stopping the client cannot be negative");
+            if (autoStopAfter < 0) {
+                throw new IllegalArgumentException("Timer for automatically stopping the client cannot be negative (value of timer: " + autoStopAfter + ")");
+            }
             this.autoStopAfter = autoStopAfter;
         }
 
-        public List<FixBean> getSessionsSettings() {
+        public List<FixBean> getSessionsSettings() throws IncorrectDataFormat {
+            Set<String> sessionIds = new HashSet<>(); // if the session IDs or session aliases are not unique, we will get an error
+            Set<String> sessionAliases = new HashSet<>();
+
+            for (FixBean fixBean : sessionsSettings) {
```

Comment:
We should also check `this` alongside this list, because `Settings` implements `FixBean` too. Or maybe we should prohibit setting session-id related fields into instance of `Settings` (this probably would be a better solution)

Reply:
mb then remove inheritance from the FixBean and add these fields separately? because prohibition of setters ==  narrowing of type. this is not really good

Reply:
It's not narrowing - getters/setters would still be accessible. If we've tried to make them protected/private that would be narrowing, but Java wouldn't allow us to do that. Overriding a getter/setter and giving it a different implementation is a valid case. If you're still oppossed to it - another option would be to remove these fields from FixBean, create its subclass which would have these fields and use this subclass as `sessionsSettings` list type